llvm/tools/opt/opt.cpp

```diff
@@ -191,6 +191,12 @@ DisableSLPVectorization("disable-slp-vectorization",
                         cl::desc("Disable the slp vectorization pass"),
                         cl::init(false));
 
+// Add this option to support the repo pruning dependent tests.
+static cl::opt<bool>
+    DisablePrepo("disable-prepo",
```

Comment:
Could you explain what this switch does? Its name suggests that it disables the repository, but its implementation seems to simply switch off the hash-generation pass. Likewise I've just noticed the "-prepo-pruning" switch...

Reply:
Yes, this switch disables the RepoMetadataGeneration (hash generation) pass. Since the RepoMetadataGeneration pass is called 'prepo' and enabled by passing '-prepo', therefore, I give the 'disable-prepo' swich to disable the RepoMetadataGeneration  pass.
To make it clear, should I change the RepoMetadataGeneration pass name to 'prepo-metadata-generation'?
I will also add another switch called 'disable-prepo-pruning' to disable the repo pruning pass. The disable punning pass will also be needed by issue 25 tests.

Reply:
Ok... Why do we need this feature?

Reply:
The repo metadata generation pass and repo pruning pass are always enabled if the build is targeted on Repo. When I add the pruning tests, the GO's digest/hash always be changed since repo metadata pass re-generated by the repo metadata. The digest changes will change the pruning status. Therefore, I added these two switches to support the repo pruning tests.

Reply:
I have removed these two switches and updated the tests.